saiph/reduction/mca.py

```diff
@@ -311,3 +312,53 @@ def stats(model: Model, df: pd.DataFrame, explode: bool = False) -> Model:
     contributions = get_variable_contributions(model, df, explode=explode)
     model.contributions = contributions
     return model
+
+
+def reconstruct_df_from_model(model: Model) -> pd.DataFrame:
+    """Reconstruct the original DataFrame from the model.
+
+    Note: if nf < df.shape[1], reconstructed df will not be exactly the same.
+
+    Parameters:
+        model: Model computed by fit.
+
+    Returns:
+        df: The reconstructed DataFrame.
+    """
+    U = model.U
+    if model.s is None:
+        raise ValueError(
+            "Model has not been fitted. Call fit() to create a Model instance."
+        )
+    S = model.s
+    Vt = model.V
+    row_w = model.row_weights
+    col_weights = model.column_weights
+    _modalities = model._modalities
+    quali = model.original_categorical
+
+    # Construct the diagonal matrix of singular values
+    Sigma = np.diag(S)
+
+    # Reconstruct the weighted and scaled matrix Z
+    Z = np.dot(U, np.dot(Sigma, Vt))
+
+    # Undo the row and column weighting
+    Z = Z / np.sqrt(row_w)[:, np.newaxis]
+    Z = Z / np.sqrt(col_weights)
+    df_reconstructed = pd.DataFrame(Z, columns=_modalities)
+
+    for var in quali:
+        prefix = var + DUMMIES_SEPARATOR
+        dummies = [col for col in df_reconstructed.columns if col.startswith(prefix)]
+        df_reconstructed[var] = (
+            df_reconstructed[dummies]
+            .idxmax(axis=1)
+            .apply(lambda x: x.split(DUMMIES_SEPARATOR)[1])
```

Comment:
why `[1]`

Reply:
To summarize, [1] is used because it accesses the second part of the split string delimited by `DUMMIES_SEPARATOR`